content/docs/reference/usage/subcommands.md

```diff
@@ -38,6 +38,32 @@ Run the program, either directly on the host or in an emulated environment (depe
 ### flash
 Flash the program to a microcontroller.
 
+The `-monitor` flag can be given to the `flash` command to start the serial
+monitor (see below) immediately after flashing. However, some microcontrollers
+need a split second or two to configure its serial port after flashing, and
+using the `-monitor` flag can fail because the serial monitor starts too
+quickly. In that case, use the `tinygo monitor` command explicitly.
+
+### monitor
+Start the serial monitor program on the serial port that is connected to the
+microcontroller.
+
+If there are multiple microcontroller attached, use the `-port` flag to select
+the serial port. On Linux, the port will be something like `/dev/ttyUSB0` or
+`/dev/ttyACM1`. On MacOS, the port will look like `/dev/cu.usbserial-1420`.
+
+The default baud rate is 115200, but can be changed using the `-baudrate` flag.
+
+The serial monitor intercepts several control characters for its own use instead
+of sending them to the microcontroller:
+
+* Control-C: terminates the `tinygo monitor`
+* Control-Z: suspends the `tinygo monitor` and drops back into shell
+* Control-\\: terminates the `tinygo monitor` with a stack trace
+* Control-S: flow-control, suspends output to the console
+* Control-Q: flow-control, resumes output to the console
+* Control-@: thrown away by `tinygo monitor`
```

Comment:
Most of these are just regular shell shortcuts, TinyGo doesn't handle them in any special way (for example, `Control-\` always terminates the current process). But I guess they could still be useful as documentation.

Reply:
I believe these are shell operations.
I think the minimum is just `ctrl-c`.
But I think it would be useful.

I would add Ctrl-J (LF) to the list.
When reading from bufio.Scanner, etc., pressing Enter (or Return or Ctrl-M : CR) does not respond. In this case, you need to press Ctrl-J (LF).

Reply:
It's been a few years since I did a deep dive into the terminal driver configuration on Linux, but my understanding is that the handling of control characters is performed by the terminal driver, not the shell. When an application launches into the foreground, it can choose to configure the terminal driver to handle these control character in a wide assortment of ways.

By default, various control characters are configured to send signals to the foreground process. For example, `^C` sends the `SIGINT`, `^Z` sends `SIGTSTP`, `^\` sends `SIGQUIT`. But the terminal driver can be configured to not send those signals at all. In addition, the terminal driver can be configured to bind alternative control characters to various signals. For example, I can run the command `$ stty intr '^X'` to replace `^C` with `^X` in the shell.

(I think there is an additional complexity in all this, because even if the terminal driver sends one of these SIGXXX signals, the application can trap it, fire off a signal handler, and then choose to either eat the control character, or pass along the control character to the microcontroller.)

Flow control seems to be slightly different than other control characters, because `^S` and `^Q` don't generate Unix signals. I believe they are handled directly in the terminal driver itself. I normally have flow control turned off in my `bash` shell, but it looks like `tinygo monitor` explicitly turns them back on, or uses some defaults with these enabled.

So I think it's useful to have these control characters listed, because `tinygo monitor` could have been configured to transparently pass every control character to the microcontroller without interpretation. But it does not. I normally use `picocom`, and I have it configured to send almost all control characters to the microcontroller, including ^C, ^Z, ^Q, ^S, and so on. Even ^\ can be configured to be ignored and sent along to the microcontroller.

@sago35: I am not sure I understand your comment about ^J, ^M, and `bufio.Scanner`. I wasn't able to find the source code for `tinygo monitor`, but I would be surprised to learn that it used buffered IO when reading from the keyboard. I *think* you are referring to the confusing complexity of handling DOS line terminators (`\r\n`) versus Unix line terminator (`\n`). To be honest, I don't know whether the `println()` or `fmt.Println()` functions send DOS terminators or Unix terminators from the microcontrollers. I have my `picocom` configured to automatically translate DOS terminators into Unix terminators, because the Arduino `Serial.println()` is usually configured to use DOS terminators. But many people write `Serial.printf("...\n")`, which sends only a Unix line terminator.

Reply:
bufio.Scanner is useful when reading lines typed from the keyboard.

In the following example, `LF` must be sent.
If you use tinygo monitor, you need to press `Ctrl-j`.

https://github.com/tinygo-org/tinygo/blob/v0.27.0/src/examples/echo2/echo2.go